#!/usr/bin/env python
import math

PACKAGE = "egocylindrical"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("width", int_t, 0, "Cylinder Height",    2048, 128, 8192)
gen.add("height", int_t, 0, "Cylinder Width",    320, 80,   1280)
gen.add("can_width", int_t, 0, "Cylinder End Widths",    default=512, min=64, max=8192)

gen.add("vfov", double_t, 0, "Cylinder Vertical FOV",    math.pi/2, .1, math.pi)
gen.add("v_offset", double_t, 0, "Cylinder Vertical shift",    0, -2, 2)

gen.add("num_threads", int_t, 0, "# threads for propagating points", 1, 1, 16)

gen.add("filter_y_min", double_t, 0, "Max y value used in filter",    -.2, -1,1)
gen.add("filter_y_max", double_t, 0, "Max y value used in filter",    .28, -1,1)

exit(gen.generate(PACKAGE, "egocylindrical", "Propagator"))
